sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java

```diff
@@ -279,99 +321,432 @@ public EventData addContext(String key, Object value) {
     }
 
     /**
-     * A collection of properties populated by Azure Event Hubs service.
+     * Gets the content type of the message.
+     *
+     * <p>
+     * Optionally describes the payload of the message, with a descriptor following the format of RFC2045, Section 5,
+     * for example "application/json".
+     * </p>
+     * @return The content type of the {@link EventData}.
+     */
+    public String getContentType() {
+        return amqpAnnotatedMessage.getProperties().getContentType();
+    }
+
+    /**
+     * Sets the content type of the {@link EventData}.
+     *
+     * <p>
+     * Optionally describes the payload of the message, with a descriptor following the format of RFC2045, Section 5,
+     * for example "application/json".
+     * </p>
+     *
+     * @param contentType RFC2045 Content-Type descriptor of the message.
+     *
+     * @return The updated {@link EventData}.
+     */
+    public EventData setContentType(String contentType) {
+        amqpAnnotatedMessage.getProperties().setContentType(contentType);
+        return this;
+    }
+
+    /**
+     * Gets a correlation identifier.
+     * <p>
+     * Allows an application to specify a context for the message for the purposes of correlation, for example
+     * reflecting the MessageId of a message that is being replied to.
+     * </p>
+     *
+     * @return The correlation id of this message.
+     */
+    public String getCorrelationId() {
+        String correlationId = null;
+        AmqpMessageId amqpCorrelationId = amqpAnnotatedMessage.getProperties().getCorrelationId();
+        if (amqpCorrelationId != null) {
+            correlationId = amqpCorrelationId.toString();
+        }
+        return correlationId;
+    }
+
+    /**
+     * Sets a correlation identifier.
+     *
+     * @param correlationId correlation id of this message
+     *
+     * @return The updated {@link EventData}.
+     * @see #getCorrelationId()
+     */
+    public EventData setCorrelationId(String correlationId) {
+        AmqpMessageId id = null;
+        if (correlationId != null) {
+            id = new AmqpMessageId(correlationId);
+        }
+        amqpAnnotatedMessage.getProperties().setCorrelationId(id);
+        return this;
+    }
+
+    /**
+     * Gets the message id.
+     *
+     * <p>
+     * The message identifier is an application-defined value that uniquely identifies the message and its payload. The
+     * identifier is a free-form string and can reflect a GUID or an identifier derived from the application context.
+     * </p>
+     *
+     * @return Id of the {@link EventData}.
+     */
+    public byte[] getUserId() {
+        return amqpAnnotatedMessage.getProperties().getUserId();
+    }
+
+    /**
+     * Sets the message id.
```

Comment:
same question. message id or user id?

Reply:
According to your comment, remove unnecessary parameters, variables and methods in the new version